### ISSUE TITLE
Camera rotated by 90° anti clockwise when coming back from an Alert Dialog

### ISSUE THREAD

**Issue by yasiralijaved** [BeyondAR/beyondar#62]
On AR Screen where Geo Objects are shown, If i open an alert dialog to let the user choose an app for Navigation or to choose an app to share his location and then if user press back button to close the Alert Dialog, the camera rotates itself by 90° anti clockwise after disappearing the alert dialog.

**Comment by IlCallo** [comment]
@joanpuigsanz 

TL;DR:
In CameraView, move the code block in from line 245 to 271 to a dedicated new method and then call it both from
`surfaceChanged()` after line 244
and 
`openCamera()` after line 401.

Long version:

I gave a look to this issue because it's pretty annoying and affect the work it's doing with this library.
I would make a pull request myself, but i have problems making the project work in android studio, so i kindly ask to you to directly change it if possible (hoping you still read this messages):

Problem seems to be that in 
CameraView line 261, `mCamera.setDisplayOrientation(orientation);`
is called just inside `surfaceChanged()` method.
This manages effectively the orientation in case of screen rotation because `surfaceChanged()` is called and, with him, 
`orientation = getCameraDisplayOrientation();` at line 258
does his work.

But if the app pause and resume without invoking surfaceChanged (eg. dialog is shown, screenshot is taken, screen is blocked and then unlocked with the activity still running), the camera is released in
BeyondarFragmentSupport `onPause()` method with command `mBeyondarCameraView.releaseCamera();` at line 178
and then open again in `onResume` method 
`mBeyondarCameraView.startPreviewCamera();` at line 167, 
which initialize the new camera object, but without ever calling surfaceChanged callback.

This happen because CameraView (which is a SurfaceView) is created just once in
BeyondarFragmentSupport `init()` method at line 159, called by `onCreateView()` of the fragment.

Given that surfaceChanged is always called after the surfaceCreated, we get that the orientation code is called the first time, but not the subsequents times when onPause/onResume take place.

To resolve this issue, i think the best thing to do is to move the code block in
CameraView `surfaceChanged()` method from line 245 to 271 to a dedicated method (`configureCameraOrientation` could be a name) and then call it both from
`surfaceChanged()` after line 244
and 
`openCamera()` after line 401.

In this way, we are sure that the orientation is checked every time we acquire the camera AND when a rotation occurs.

BONUS (minor issues i'm not sure about):
1) In CameraView, `acquiredCam = true;` command at line 403 is useless (and moreover the variable itself is useless) because just after that assignment we return true, so the while cycle ends anyway.
The variable acquiredCam can be removed and while can cycle over `while(timePassed < 1000)` and the effect would be the same.

2) I think there can also be a problem with the fact that every time that in
CameraView `init()` method, with command `this.mHolder.addCallback(this);` at line 97
we add a callback, we are adding more and more callbacks references to the same SurfaceView (the current one), because that method can hold multiple Callback interface.
I'm not sure about this because i don't know how does SurfaceHolder manage/stores those callbacks reference internally, maybe he already checks for them to be references to different interfaces.
